engine/build.gradle

```diff
@@ -244,3 +244,8 @@ idea {
         downloadSources = true
     }
 }
+
+// Make sure our config file for code analytics get extracted (vulnerability: non-IDE execution of single analytic)
+ideaModule.dependsOn rootProject.extractConfig
+tasks.eclipse.dependsOn rootProject.extractConfig
+check.dependsOn rootProject.extractConfig
```

Comment:
Maybe the reason for the slowdown is that the extraction of file from the zip archive is done in the configuration phase?

In any case, it might be safer to make the root projects `checkstyleMain` and `checkstyleTest` tasks depend on the extraction. This should be fast and reliable, but fails if someone cleans and runs checkstyle on engine only. This seems to be a less serious issue to me.